docs/docs/packaging.md

```diff
@@ -85,3 +85,63 @@ With conda, dependency constraints of the form `X.Y.*` will permit the installat
 pip is stricter and will not generally install nightlies unless the `--pre` argument is specified.
 However, pip can be instructed to install nightlies of specific packages by [using any specifier containing a pre-release or development version](https://pip.pypa.io/en/stable/cli/pip_install/#pre-release-versions).
 Therefore, during wheel builds we also do this [in our CI scripts by adding a `>=0.0.0a0` specifier to all the relevant constraints](https://github.com/rapidsai/cudf/blob/branch-24.06/ci/build_wheel.sh#L34).
+
+## `dependencies.yaml` Conventions
+
+Dependency lists (under the `dependencies` top-level key) shall have the
+following conventions:
+
+- Build dependencies (under `build-system.requires`) shall be under a
+  dependency list named `py_build_<wheel_name>`.
+- RAPIDS build backend dependencies (under
+  `tool.rapids-build-backend.requires`) shall be under a dependency list named
+  `py_rapids_build_<wheel_name>`.
```

Comment:
```suggestion
- Build-time dependencies intended for the `build-system.requires` table shall be under a
  dependency list named `py_build_<wheel_name>`. This should only be packages
  that are required for `rapids-build-backend` to run and generate wheel metadata. For example:
    - `rapids-build-backend` itself
    - whatever provides the build backend indicated in `tool.rapids-build-backend.build-backend` (e.g.`scikit-build-core` or `setuptools`)
    - if the project uses a `setup.py`, and libraries that are imported within that `setup.py`
- Build-time dependencies intended for the `tool.rapids-build-backend.requires` table
  shall be under a dependency list named `py_rapids_build_<wheel_name>`. This should be
  packages that are needed to create the files that end up in the wheel. For example:
    - build tools like `ninja` and `cmake`
    - `Cython`
    - packages providing headers / libraries used during compilation (e.g. `librmm`, `libcudf`)
```

I think the phrase "RAPIDS build backend dependencies" here is confusing. I could see that being interpreted as "the things required to run `rapids-build-backend`", and I also think that to someone who isn't already very familiar with this setup, the difference between "RAPIDS build backend dependencies" and "Build dependencies" would not be clear.

Would you consider something like this proposed revision, that separates these two lists by *purpose* and provides some examples?

Reply:
Applied your suggestion.